src/main/java/io/sprucehill/zalando/api/nativecart/service/AccessTokenService.java

```diff
@@ -17,15 +24,33 @@
 
 	@Override
 	public TokenInfo read() throws Exception {
-		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientCredential);
-		request.setHeader("Content-Type","application/x-www-form-urlencoded");
+		HttpPost request = postRequest("/access_token");
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
+		
+		List<NameValuePair> params = new ArrayList<NameValuePair>();
+		params.add(new BasicNameValuePair("realm","%2Fcustomers"));
+		params.add(new BasicNameValuePair("grant_type","client_credentials"));
+		params.add(new BasicNameValuePair("client_id",clientId));
+		params.add(new BasicNameValuePair("client_secret",clientCredential));
+		
+		request.setEntity(new UrlEncodedFormEntity(params));
+		
 		return execute(request, new TypeReference<TokenInfo>() {});
 	}
 	
 	@Override
 	public TokenInfo readFromAuthorizationCode(String authCode,String redirectUrl) throws Exception {
 		HttpPost request = postRequest("/access_token?realm=%2Fcustomers&grant_type=authorization_code&client_id="+clientId+"&client_secret="+clientCredential+"&code="+authCode+"&redirect_uri="+redirectUrl);
-		request.setHeader("Content-Type","application/x-www-form-urlencoded");
+		request.setHeader(HttpHeaders.CONTENT_TYPE,"application/x-www-form-urlencoded");
```

Comment:
You might not need this here when using `UrlEncodedFormEntity` - this entity class should add the correct `Content-Type` header itself